docs/framework/civimail.md

```diff
@@ -0,0 +1,155 @@
+
+For more information see this [Stack Exchange Question](https://civicrm.stackexchange.com/questions/2314/how-does-civimail-reply-tracking-and-forwarding-work).
+
+## Jobs
+
+### Queuing
+
+Given a job, find all email addresses that should receive the mailing (`job.mailing_id`). 'email address' is taken to mean the primary email of the primary location of a contact, unless one (or both) of the email and location fields in the contact-group join table are non-null.  For inclusion, only contacts where `do_not_email = 0` and `on_hold = 0` are considered.
+
+If a job has `is_retry = 1`, the queue is generated by finding all jobs with the same `mailing_id` and joining to bounce event (through queue). Otherwise, the queue is generated as follows:
+
+- Included recipients (`group_type = Include`) 
+    - Static groups - All contacts belonging to (`status = In`) any group in `crm_mailing_group where mailing_id = job.mailing_id`. 
+    - Saved searches - All contacts matching the where clause of a saved search linked from groups in `crm_mailing_group`.
+    - Previous mailings - All email addresses in `Mailing_Event_Queue` with the `job.mailing_id` keyed from the mailing-group join table. Only jobs with `Complete` status should be considered.
+
+- Excluded recipients (`group_type = Exclude`)
+    - Static groups
+        - All contacts belonging to (`status = In`) any group in `crm_mailing_group where mailing_id = job.mailing_id`.
+        -   Saved searches - All contacts matching the where clause of a saved search linked from groups in `crm_mailing_group`.
+        -   Previous mailings -   All email addresses in `Mailing_Event_Queue` with the `job.mailing_id` keyed from the mail-group join table.
+        -   Successful deliveries in previous jobs of the same mailing  -   All email addresses in `Mailing_Event_Queue` where `job.mailing_id = mailing_id` inner join to `Mailing_Event_Delivery` and left join to `Mailing_Event_Bounce` where `Mailing_Event_Delivery.id <> null` and `Mailing_Event_Bounce.id is null`.
+
+### Status
+
+Job `status` can take one of 5 states.
+
+1.  `Scheduled`: All jobs start in this state, and stay there until the `start_date` has passed.
+2.  `Running`: Jobs are marked `Running` only after the entire recipient queue (`Mailing_Event_Queue`) has been constructed. If the queuing process is interrupted, the job will remain in `Scheduled` and the queue will be reconstructed the next time the mailer is run.
+3.  `Complete`: Jobs are marked complete after the mailer has attempted to send the message to every recipient in the queue (ie. every queue event has a corresponding bounce or delivery event). Note that slow bounce events may continue well after the job has been marked `Complete`.
+4.  `Paused`: A job can only be marked paused by the admin interface. The mailer will not act on paused jobs.
+5.  `Canceled`: Like paused, but cannot be placed back in the `Running` state.
+
+## Events
```

Comment:
Not sure "events" is quite the right name here, but not sure what else to suggest.  May also be useful to note somewhere here that an activity is created against each contact with a link to the mailing